views/js/controller/Booklet/editBooklet.js

```diff
@@ -0,0 +1,71 @@
+/*  
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ * 
+ * Copyright (c) 2008-2010 (original work) Deutsche Institut für Internationale Pädagogische Forschung (under the project TAO-TRANSFER);
+ *               2009-2012 (update and modification) Public Research Centre Henri Tudor (under the project TAO-SUSTAIN & TAO-DEV);
+ * 
+ */
+define(['jquery', 'lodash', 'module', 'helpers', 'generis.tree.select'], function ($, _, module, helpers, GenerisTreeSelectClass) {
+    'use strict';
+
+
+    function buildGroupsTree(container) {
+
+        var treeOptions = container.data();
+
+        new GenerisTreeSelectClass('#' + treeOptions.actionId + '-tree', helpers._url('getData', 'GenerisTree', 'tao'), {
+            actionId: treeOptions.actionId,
+            saveUrl: treeOptions.saveUrl,
+            saveData: {
+                resourceUri: treeOptions.resourceUri,
+                propertyUri: treeOptions.propertyUri
+            },
+            checkedNodes: treeOptions.checkedNodes,
+            serverParameters: {
+                openNodes: treeOptions.openNodes,
+                rootNode: treeOptions.rootNode
+            },
+            saveCallback: function () {
+                helpers._load(
```

Comment:
This function is depreacted, use the section API instead. See `tao/views/js/layout/section.js` 
For example 

```javascript
section
  .current()
   .loadContentBlock(
        helpers._url('editBooklet', 'Booklet', 'taoBooklet'), 
        {uri: treeOptions.resourceUri}
    );
```